execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -620,38 +775,80 @@ where
 
         let mut named_keys = NamedKeys::new();
 
-        let genesis_validators: BTreeMap<PublicKey, U512> = self
-            .exec_config
-            .accounts()
-            .iter()
-            .filter_map(|genesis_account| {
-                if genesis_account.is_genesis_validator() {
-                    Some((
-                        genesis_account
-                            .public_key()
-                            .ok_or(GenesisError::MissingPublicKey)
-                            .ok()?,
-                        genesis_account.bonded_amount().value(),
-                    ))
-                } else {
-                    None
-                }
-            })
-            .collect();
+        let genesis_validators: Vec<_> = self.exec_config.get_bonded_validators().collect();
+
+        let genesis_delegators: Vec<_> = self.exec_config.get_bonded_delegators().collect();
```

Comment:
There's probably an opportunity to optimize to a single iteration here, but given that code is being run once at genesis, this seems acceptable for now.